server/app/repository/ApplicationStatusesRepository.java

```diff
@@ -52,4 +52,22 @@ public List<StatusDefinitions> lookupListOfObsoleteStatusDefinitions(String prog
         .map(ApplicationStatusesModel::getStatusDefinitions)
         .collect(Collectors.toList());
   }
+
+  public void updateStatusDefinitions(String programName, StatusDefinitions statusDefinitions) {
+    // archive the previous active one
+    database
+        .update(ApplicationStatusesModel.class)
+        .set("status_definitions_lifecycle_stage", StatusDefinitionsLifecycleStage.OBSOLETE)
+        .where()
+        .eq("program_name", programName)
+        .and()
+        .eq("status_definitions_lifecycle_stage", StatusDefinitionsLifecycleStage.ACTIVE)
+        .update();
+
+    // create new status
+    ApplicationStatusesModel newStatusDefinition =
+        new ApplicationStatusesModel(
+            programName, statusDefinitions, StatusDefinitionsLifecycleStage.ACTIVE);
+    newStatusDefinition.save();
```

Comment:
**issue**: Can we wrap this in a transaction so it happens atomically. otherwise a database read between these two writes could end up with unexpected data, like there not being an ACTIVE `StatusDefinitions` for a program.

Reply:
https://ebean.io/docs/transactions/

Reply:
Great idea. Done!